### PR TITLE
feat: browser uploaded file in ckeditor

### PR DESCRIPTION
**Pull request by IsaMic**
|Q              |A  |
|---------------|---|
|Bug fix?       |N|
|New feature?   |Y|	
|BC breaks?     |N|	
|Deprecations?  |N|	
|Fixed tickets  |N|	

Need to select uploaded files on server in wysiwyg, add a browser in adv_link of Ckeditor plugin,

!!! Use new fonction and const in common bundle : https://github.com/ems-project/EMSCommonBundle/pull/259 => 3 errors PHPStan

### CODE REVIEW

File: src/Resources/views/ajax/files.json.twig [hunk 1]
```diff
@@ -0,0 +1,8 @@
+[{% for file in files %}
+	{
+		"file": {{ ('ems://asset:'~ file.sha1 ~ '?name='~ file.name ~ '&type=' ~ file.type)|json_encode|raw}},
+		"name": {{ file.name|json_encode|raw }},
+		"type": {{ file.type|json_encode|raw }},
+		"folder": {{ file.user|json_encode|raw }}
+	}{% if not loop.last %},{% endif %}{% endfor %}
+]
```
Comment (theus77) [inline src/Resources/views/ajax/files.json.twig:8]: Build a files twig variables than `{{ files|json_encode|raw }}` or directly returns a jsonresponse from the controller without any json

File: src/Resources/public/js/cke-plugins/adv_link/browser/jquery-1.9.1.min.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+/*! jQuery v1.9.1 | (c) 2005, 2012 jQuery Foundation, Inc. | jquery.org/license
+//@ sourceMappingURL=jquery.min.map
```
Comment (theus77) [inline src/Resources/public/js/cke-plugins/adv_link/browser/jquery-1.9.1.min.js:2]: Must be removed

File: assets/cke-plugins/adv_link/browser/jquery-1.9.1.min.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+/*! jQuery v1.9.1 | (c) 2005, 2012 jQuery Foundation, Inc. | jquery.org/license
```
Comment (theus77) [inline assets/cke-plugins/adv_link/browser/jquery-1.9.1.min.js:1]: This files must be removed.

File: src/Controller/UploadedFileWysiwygController.php [hunk 4]
```diff
@@ -0,0 +1,83 @@
+<?php
+
```
Comment (theus77) [inline src/Controller/UploadedFileWysiwygController.php:2]:
```suggestion

declare(strict_types=1);
```

File: src/Controller/UploadedFileWysiwygController.php [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+use EMS\CoreBundle\Form\Form\TableType;
+use EMS\CoreBundle\Helper\DataTableRequest;
+use EMS\CoreBundle\Service\FileService;
+use Psr\Log\LoggerInterface;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+class UploadedFileWysiwygController extends AbstractController
```
Comment (theus77) [inline src/Controller/UploadedFileWysiwygController.php:20]:
```suggestion
final class UploadedFileWysiwygController extends AbstractController
```

File: src/Controller/UploadedFileWysiwygController.php [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+        return $this->render('@EMSCore/uploaded-file-wysiwyg/index.html.twig', [
+            'form' => $form->createView(),
+            'CKEditorFuncNum' => $request->query->get('CKEditorFuncNum') ?: 0,
+        ]);
+    }
+
+    private function initTable(): EntityTable
+    {
+        $table = new EntityTable($this->fileService, $this->generateUrl('ems_core_uploaded_file_ajax'));
+        $table->addColumnDefinition(new BoolTableColumn('uploaded-file.index.column.available', 'available'));
```
Comment (theus77) [inline src/Controller/UploadedFileWysiwygController.php:59]: You should only load available files

File: src/Service/FileService.php [hunk 7]
```diff
@@ -372,6 +372,10 @@ public function isSortable(): bool
 
     public function get(int $from, int $size, ?string $orderField, string $orderDirection, string $searchValue, $context = null): array
     {
+        if (null != $context && isset($context['available']) && !$context['available']) {
+            return $this->uploadedAssetRepository->getAvailable($from, $size, $orderField, $orderDirection, $searchValue);
+        }
+
```
Comment (theus77) [inline src/Service/FileService.php:378]:
```suggestion
        if (null !== $context && ($context['available'] ?? false)) {
            return $this->uploadedAssetRepository->getAvailable($from, $size, $orderField, $orderDirection, $searchValue);
        }

```

**Review by theus77 — APPROVED** [review]

**Review by Davidmattei — APPROVED** [review]

**Pull request merged**